deepchem/models/tensorgraph/models/graph_models.py

```diff
@@ -600,28 +609,47 @@ def evaluate(self, dataset, metrics, transformers=[], per_task_metrics=False):
         labels=self.my_labels,
         weights=[self.my_task_weights])
 
-  def predict_on_smiles(self, smiles, transformers):
-    max_index = len(smiles)
-    num_batches = max_index // self.batch_size
+  def bayesian_predict(self, X, transformers=[], n_passes=4, untransform=False):
```

Comment:
Could you add a docstring here?

Also, instead of accepting a numpy array `X`, this should accept either a dataset object or smiles.